src/DevBetterWeb.Web/CustomAttributes/ValidCodinGameUrlAttribute.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System;
+using System.ComponentModel.DataAnnotations;
+
+namespace DevBetterWeb.Web
+{
+  public class ValidCodinGameUrlAttribute : ValidationAttribute
+  {
+    public string GetErrorMessage() =>
+    $"You must enter a valid CodinGame URL";
+
+    protected override ValidationResult IsValid(object value,
+        ValidationContext validationContext)
+    {
+      if (value == null)
+      {
+        return ValidationResult.Success;
+      }
+
+      var url = value.ToString();
+
+      if (string.IsNullOrEmpty(url))
+      {
+        return ValidationResult.Success;
+      }
+
+
+      var isValueUrl = Uri.TryCreate(url, UriKind.Absolute, out Uri? uri);
+
+      var isCodinGameUrl = url.ToLower().Contains("codingame");
```

Comment:
Instead of one attribute per web site, couldn't you create a `ValidUrlContainingStringAttribute` and just pass in the staring you're saying it needs to have?

Sample usage:

```
[ValidUrlContainingString("codingame.com")]
```

Reply:
sighhh good point